core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -406,6 +406,13 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       .map { case (k, v) => (k.substring(prefix.length), v) }
   }
 
+  /**
+   * Get all parameters that start with `prefix` and end with 'suffix'
+   */
+  def getAllWithPrefixAndSuffix(prefix: String, suffix: String): Array[(String, String)] = {
```

Comment:
I saw this method is frequently used but I'm not sure this is the right home for it. Even `object SparkConf` is package private, to me those methods seem too specific and they will show up in Java API docs. The return result is also very specific to our use case. cc: @gatorsmile for suggestions

Reply:
I put it here because it really seems like just a variation of getAllWithPrefix. There are a bunch of configs now that use this prefix type format, but if you have a better idea I can move it and we can always propogate it up if some one else needs it.